lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/classify/stats/CubePercentile.java

```diff
@@ -0,0 +1,60 @@
+package edu.stanford.futuredata.macrobase.analysis.classify.stats;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+
+import java.util.*;
+
+/**
+ * Computes percentiles when given an array of metrics and the count of the
+ * number of times each occurs. Useful for computing percentiles on cubed data.
+ */
+public class CubePercentile {
+    private double[] counts;
+    private double[] metrics;
+
+    // Computed
+    private int numRawMetrics = 0;
+    private Multiset<Double> metricCounts;
```

Comment:
Multiset will be pretty slow when you have groups with large counts: remember the big advantage of using cubes is to minimize memory usage and time by not having the explode out the raw rows.

Instead of metricCounts and sortedMetrics we should try something like WeightedMetric[] metrics. where WeightMetric is a metric value and a count/weight.

To see this in action try creating a unit test where you have cubes with only 2 dimensions and say counts of 1Million: this code will be relatively slow in that case when it could be way faster if you don't explode out the counts.